src/lib.rs

```diff
@@ -161,6 +162,16 @@ impl Mutagen {
             _ => x >= y,
         }
     }
+
+    pub fn forloop<'a, I: Iterator + 'a>(&self, i: I, n: usize) -> Box<Iterator<Item=I::Item> + 'a> {
```

Comment:
If I'm not wrong, I can't use the `impl Trait` notation as it's meant to be used for static dispatch (at least, I couldn't make it work).
I don't know if there's any work around to avoid boxing the iterator.